pkg/api/proxy.go

```diff
@@ -22,6 +22,8 @@ import (
 	"github.com/compose-spec/compose-go/types"
 )
 
+var _ Service = (*ServiceProxy)(nil)
```

Comment:
Just curious: any reason for this change? The old way (`var _ Service = &ServiceProxy{}`) is the way I've always seen it in Go code, and won't actually result in any allocations/memory usage at runtime.

Reply:
Nothing special... Just that it visually avoids allocation. But it makes total sense that the compiler just avoids it in an optimisation phase.
I'll change it back then. The classical one looks prettier.

Reply:
Ahh yeah I def agree that the "idiomatic" way is 🪄 magic 🪄